site/component/manifests/redis.pp

```diff
@@ -1,3 +1,15 @@
-class component::redis {
-  package { 'redis-server': }
+class component::redis($keep_bc = true, $instances = {}) {
+  validate_bool($keep_bc)
+  validate_hash($instances)
+
+  if $keep_bc {
+    package { 'redis-server': }
```

Comment:
Why is backwards compatibility needed? Nothing should depend on the package itself.